src/Windows/R/Editor/Impl/SuggestedActions/RSuggestedActionsSource.cs

```diff
@@ -0,0 +1,103 @@
+    internal sealed class RSuggestedActionsSource : SuggestedActionsSourceBase, ISuggestedActionsSource {
+        private readonly IREditorDocument _document;
+        private IAstNode _lastNode;
+
+        private RSuggestedActionsSource(ITextView textView, ITextBuffer textBuffer, IEnumerable<ISuggestedActionProvider> suggestedActionProviders, IServiceContainer services) :
+            base(textView, textBuffer, suggestedActionProviders, services) {
+            _document = (IREditorDocument)Document;
+        }
+
+        public static ISuggestedActionsSource Create(ITextView textView, ITextBuffer textBuffer, IServiceContainer services) {
+            // Check for detached documents in the interactive window projected buffers
+            var document = textBuffer.GetEditorDocument<IREditorDocument>();
+            if (document == null || document.IsClosed) {
+                return null;
+            }
+            var suggestedActionProviders =
+                ComponentLocator<ISuggestedActionProvider>.ImportMany(services.GetService<ICompositionService>()).Select(p => p.Value);
+            return new RSuggestedActionsSource(textView, textBuffer, suggestedActionProviders, services);
+        }
+
+        protected override void OnCaretPositionChanged(object sender, CaretPositionChangedEventArgs e) {
+            if (!IsDisposed && _document?.EditorTree != null && !_document.IsClosed) {
+                var bufferPoint = TextView.GetCaretPosition(_document.TextBuffer());
+                if (bufferPoint.HasValue) {
+                    var node = _document.EditorTree.AstRoot.GetNodeOfTypeFromPosition<TokenNode>(bufferPoint.Value);
+                    if (node != _lastNode) {
+                        _lastNode = node;
+                        SuggestedActionsChanged?.Invoke(this, EventArgs.Empty);
+                    }
+                }
+            }
+        }
+
+        #region ISuggestedActionsSource
+        public event EventHandler<EventArgs> SuggestedActionsChanged;
+
+        public IEnumerable<SuggestedActionSet> GetSuggestedActions(ISuggestedActionCategorySet requestedActionCategories, SnapshotSpan range, CancellationToken cancellationToken) {
+            if (IsDisposed ||
+                cancellationToken.IsCancellationRequested ||
+                !range.Snapshot.TextBuffer.ContentType.TypeName.EqualsOrdinal(RContentTypeDefinition.ContentType)) {
+                return Enumerable.Empty<SuggestedActionSet>();
+            }
+
+            var caretPosition = TextView.Caret.Position.BufferPosition;
+            var bufferPoint = TextView.MapDownToR(caretPosition);
+            if (!bufferPoint.HasValue) {
+                return Enumerable.Empty<SuggestedActionSet>();
+            }
+
+            var ast = _document?.EditorTree?.AstRoot;
+            var bufferPosition = bufferPoint.Value.Position;
+            _lastNode = ast?.GetNodeOfTypeFromPosition<TokenNode>(bufferPosition);
```

Comment:
Probably not. This is call to actually retrieve list of actions for the given position so wherever the caret position is - is the last node.